### PR TITLE
Add datetime

### PR DESCRIPTION
**Pull request by plynchnlm**

### CODE REVIEW

File: test/fhirpath.test.js [hunk 1]
```diff
@@ -9,15 +9,15 @@ const _    = require('lodash');
 const items = fs.readdirSync(__dirname + '/cases/');
 
 // Set "focus" to true to turn on focus option
+const focus = false;
+const focusFile = /.*.yaml/;
+
 
 const endWith = (s, postfix) => {
   const idx = s.indexOf(postfix);
   return (idx > 0 && idx === s.length - postfix.length);
```
Comment (xluanlhc) [inline test/fhirpath.test.js]: If there are multiple occurrences of postfix in s? Maybe: s.length >= postfix.length && s.subtr(s.length - postifx.lenth) === postfix?

File: src/datetime.js [hunk 2]
```diff
@@ -1,10 +1,67 @@
...
+ *  3. (Optional-- default is 2).
+ */
+function formatNum(num, len) {
+  // Could use String.repeat, but that requires convertin num to an string first
+  // to get its length.  This might be slightly faster given that we only need 2
+  // or three 3 digit return values.
+  var rtn = num;
+  if (len === 3 && num < 100)
+    rtn = '0' + num;
+  // else assume len === 2
```
[the remark this answers is not in the document]
Comment (plynchnlm) [reply src/datetime.js]: Removed.

File: src/equality.js [hunk 3]
```diff
@@ -31,42 +37,76 @@ engine.unequival = function(a, b){
   return !equivalence(a, b);
 };
 
+/**
+ *  Checks that the types of a and b are suitable for comparison in an
+ *  inequality expression.
+ * @param a the left side of the inequality expression (which should an array of
```
[the remark this answers is not in the document]
Comment (plynchnlm) [reply src/equality.js]: Fixed.

File: src/equality.js [hunk 4]
```diff
@@ -31,42 +37,76 @@ engine.unequival = function(a, b){
...
+  let rtn = null;
   util.assertAtMostOne(a, "Singleton was expected");
   util.assertAtMostOne(b, "Singleton was expected");
   a = a[0];
   b = b[0];
-  let lType = typeof a;
-  let rType = typeof b;
-  if (lType != rType) {
-    util.raiseError('Type of "'+a+'" did not match type of "'+b+'"', 'InequalityExpression');
+  let lClass = a.constructor;
```
[the remark this answers is not in the document]
Comment (plynchnlm) [reply src/equality.js]: The check that a and b have at least one value has already been made prior to this function being called.  I added a comment to clarify.

File: src/misc.js [hunk 5]
```diff
@@ -67,4 +70,28 @@ engine.toString = function(coll){
...
 };
 
+
+/**
+ *  Defines a function on engine called to+timeType (e.g., toDateTime, etc.).
+ * @param timeType a class (contsructor) for a time type (e.g. DateTime).
+ */
+function defineTimeConverter(timeType) {
+  let timeName = timeType.name;
+  if (timeName.indexOf('FP_') === 0)
```
[the remark this answers is not in the document]
Comment (plynchnlm) [reply src/misc.js]: Good point.  Removed that check, and corrected comment.

File: test/fhirpath.test.js (hunk 1 above)
Comment (plynchnlm) [reply test/fhirpath.test.js]: Good catch.  Fixed.

**Pull request merged**